### ISSUE TITLE
[terra-modal-manager] Screen reader issue

### ISSUE THREAD

**Issue by ShettyAkarsh** [cerner/terra-framework#1210]
With screen reader ON . if user tries to launch the modal through [examples](https://engineering.cerner.com/terra-ui/components/terra-modal-manager/modal-manager/about) provided in terra-ui. As the dialog launches the screen reader reads it as 'Modal, Dialog ...." as we are setting the arial-label value [here](https://github.com/cerner/terra-framework/blob/main/packages/terra-modal-manager/src/ModalManager.jsx#L96)  to 'Modal" always.

so issue here is shouldn't  the value to aria-label be passed by the consumers or it should be equivalent to the [title](https://github.com/cerner/terra-framework/blob/main/packages/terra-modal-manager/src/ModalManager.jsx#L104) prop value of action header. ? so that the screen readers reads out the appropriate text when the modal opens.
